lib/Controller/ItemApiController.php

```diff
@@ -166,6 +166,22 @@ private function setStarred($isStarred, $feedId, $guidHash)
     }
 
 
+    private function setStarredById($isStarred, $itemId)
+    {
+        try {
+            $this->itemService->starById(
+                $itemId,
+                $isStarred,
+                $this->getUserId()
+            );
```

Comment:
I'd rather this be a `star(string $user_id, array $item_ids, bool $starred)` in the ItemServiceV2

Reply:
@SMillerDev does ItemServiceV2 provide the means to fetch news items by ID? V1 mapper used in V1 service has a convenient function `find(string $userId, int $id)`. I don't see anything like that in the V2 mapper.

Reply:
It's findForUser

Reply:
@SMillerDev if you mean `findFromUser`, I tried this one yesterday and it doesn't seem to work. Here is the stack trace:

```
#1 /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Query/QueryBuilder.php(1167): Doctrine\DBAL\Query\QueryBuilder-&gt;verifyAllAliasesAreKnown(Array)
#2 /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Query/QueryBuilder.php(1127): Doctrine\DBAL\Query\QueryBuilder-&gt;getFromClauses()
#3 /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Query/QueryBuilder.php(244): Doctrine\DBAL\Query\QueryBuilder-&gt;getSQLForSelect()
#4 /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Query/QueryBuilder.php(206): Doctrine\DBAL\Query\QueryBuilder-&gt;getSQL()
#5 /var/www/html/lib/private/DB/QueryBuilder/QueryBuilder.php(217): Doctrine\DBAL\Query\QueryBuilder-&gt;execute()
#6 /var/www/html/lib/public/AppFramework/Db/QBMapper.php(262): OC\DB\QueryBuilder\QueryBuilder-&gt;execute()
#7 /var/www/html/lib/public/AppFramework/Db/QBMapper.php(346): OCP\AppFramework\Db\QBMapper-&gt;findOneQuery(Object(OC\DB\QueryBuilder\QueryBuilder))
#8 /var/www/html/custom_apps/news/lib/Db/ItemMapperV2.php(96): OCP\AppFramework\Db\QBMapper-&gt;findEntity(Object(OC\DB\QueryBuilder\QueryBuilder))
#9 /var/www/html/custom_apps/news/lib/Service/ItemServiceV2.php(128): OCA\News\Db\ItemMapperV2-&gt;findFromUser(&#039;admin&#039;, 2)
#10 /var/www/html/custom_apps/news/lib/Controller/ItemApiController.php(323): OCA\News\Service\ItemServiceV2-&gt;starById(2, true, &#039;admin&#039;)
#11 /var/www/html/custom_apps/news/lib/Controller/ItemApiController.php(354): OCA\News\Controller\ItemApiController-&gt;setMultipleStarredById(true, Array)
#12 /var/www/html/lib/private/AppFramework/Http/Dispatcher.php(169): OCA\News\Controller\ItemApiController-&gt;starMultipleById(Array)
#13 /var/www/html/lib/private/AppFramework/Http/Dispatcher.php(100): OC\AppFramework\Http\Dispatcher-&gt;executeController(Object(OCA\News\Controller\ItemApiController), &#039;starMultipleByI...&#039;)
```

Reply:
It works in other parts of the app. What's the reason for the stacktrace?

Reply:
@SMillerDev 

```
Message: The given alias &#039;`items`&#039; is not part of any FROM or JOIN clause table. The currently registered aliases are: `*PREFIX*news_items`.
```

V1 mapper works fine with the same args

Reply:
Ah, I was mixing up my naming. https://github.com/nextcloud/news/blob/master/lib/Service/Service.php#L97 and https://github.com/nextcloud/news/blob/master/lib/Service/ItemServiceV2.php#L75 should be all you need.

Reply:
@SMillerDev It looks like this method is also broken. Here is the error:

```
Message: Call to undefined method OCA\News\Db\ItemMapperV2::find()
```

Here is the full code:

https://github.com/nextcloud/news/blob/8b670a6e239375133c3f8c0e66ce31fcb187cc3b/lib/Service/ItemServiceV2.php#L113-L124